cli/commands/checkpoint.go

```diff
@@ -0,0 +1,110 @@
+package commands
+
+import (
+	"context"
+
+	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/core"
+	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/core/onchain"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/fatih/color"
+	"github.com/pkg/errors"
+)
+
+type TCheckpointCommandArgs struct {
+	EigenpodAddress     string
+	Node                string
+	BeaconNode          string
+	Sender              string
+	DisableColor        bool
+	NoPrompt            bool
+	SimulateTransaction bool
+	BatchSize           uint64
+	ForceCheckpoint     bool
+	Verbose             bool
+}
+
+func CheckpointCommand(args TCheckpointCommandArgs) error {
+	ctx := context.Background()
+
+	if args.DisableColor {
+		color.NoColor = true
+	}
+
+	isVerbose := !args.SimulateTransaction || args.Verbose
+
+	if args.SimulateTransaction && len(args.Sender) > 0 {
+		core.Panic("if using `--print-calldata`, please do not specify a sender.")
+		return nil
+	}
+
+	eth, beaconClient, chainId, err := core.GetClients(ctx, args.Node, args.BeaconNode, isVerbose)
+	core.PanicOnError("failed to reach ethereum clients", err)
+
+	currentCheckpoint, err := core.GetCurrentCheckpoint(args.EigenpodAddress, eth)
+	core.PanicOnError("failed to load checkpoint", err)
+
+	eigenpod, err := onchain.NewEigenPod(common.HexToAddress(args.EigenpodAddress), eth)
+	core.PanicOnError("failed to connect to eigenpod", err)
+
+	if currentCheckpoint == 0 {
+		if len(args.Sender) > 0 || args.SimulateTransaction {
+			if !args.NoPrompt && !args.SimulateTransaction {
+				core.PanicIfNoConsent(core.StartCheckpointProofConsent())
+			}
```

Comment:
Can we do the inverse of this boolean statement?

`if args.NoPrompt || args.SimulateTransaction`

The current version is making my head hurt :(